libs/island-ui/core/src/lib/TableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FC } from 'react'
+import { Text } from '../Text/Text'
+import { Box } from '../Box/Box'
+import { Stack } from '../Stack/Stack'
+
+interface Props {
+  tableOfContentsTitle: string
+  headings: Array<{ headingTitle: string; headingId: string }>
+  onClick: (selectedHeadingId: string) => void
+}
+
+export const TableOfContents: FC<Props> = ({
+  tableOfContentsTitle,
+  headings,
+  onClick,
+}) => (
+  <Box
+    paddingX={4}
+    paddingY={2}
+    borderLeftWidth="standard"
+    borderColor="blue200"
+  >
+    <Stack space={1}>
+      <Text variant="h5" as="h5">
+        {tableOfContentsTitle}
+      </Text>
+      {headings.map(({ headingTitle, headingId }, id) => (
+        <Box
+          key={id}
+          component="button"
+          type="button"
+          textAlign="left"
+          onClick={() => onClick(headingId)}
+        >
```

Comment:
maybe use reakit button instead of Box, I'm not totally sure if we need to adjust the styling but it will provide us with some nice accessibility features